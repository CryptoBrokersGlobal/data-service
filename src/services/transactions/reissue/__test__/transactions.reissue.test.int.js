const createService = require('..');
const { createPgDriver } = require('../../../../db');

const TX_ID = '2JksHUA6E4ZQd351i3oxJq2Ts7tvMN7NNVfoduR1fFsk';
const TX_ID_2 = '2TyaLLCzjvC3tVTWebq5hAvEakXeACRrf3Bgu4e8fSos';

const loadConfig = require('../../../../loadConfig');
const options = loadConfig();

const drivers = {
  pg: createPgDriver(options),
};

const service = createService({
  drivers,
  emitEvent: () => () => null,
});

describe('Reissue transaction service', () => {
  describe('get', () => {
    it('fetches real tx', async done => {
      service
        .get(TX_ID)
        .run()
        .promise()
        .then(x => {
          expect(x).toMatchSnapshot();
          done();
        })
        .catch(e => done(JSON.stringify(e)));
    });
    it('returns null for unreal tx', async () => {
      const tx = await service
        .get('UNREAL')
        .run()
        .promise();

      expect(tx).toBe(null);
    });
  });

  describe('mget', () => {
    it('fetches real txs with nulls for unreal', async done => {
      service
        .mget([TX_ID, 'UNREAL', TX_ID_2])
        .run()
        .promise()
        .then(xs => {
          expect(xs).toMatchSnapshot();
          done();
        })
        .catch(e => done(JSON.stringify(e)));
    });
  });
});
