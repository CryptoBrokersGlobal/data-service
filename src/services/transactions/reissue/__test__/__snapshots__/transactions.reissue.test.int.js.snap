// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reissue transaction service get fetches real tx 1`] = `
Object {
  "__type": "transaction",
  "data": Object {
    "assetId": "F7CrEp9tihJSxTAAv2Kai1QQWBdF2fp3tyiDMRuvtcQm",
    "fee": "1",
    "height": 263697,
    "id": "2JksHUA6E4ZQd351i3oxJq2Ts7tvMN7NNVfoduR1fFsk",
    "quantity": "500",
    "reissuable": false,
    "sender": "3P7HKYe6HB8eK7uXxrztEYUSzZQyyYRnWHg",
    "senderPublicKey": "9ASL55YRs78rW5hjRAgjyWww7QqP99HwuB8dG6AqtnFm",
    "signature": "3FsqtYU1gt51F4YyEpG43e9XntY3qkxUdUZzxMzvb28MpaxytE7pw22gafwe11Za3uLvDvWUcxnoHj1dqCtRwfhE",
    "timestamp": 2016-12-06T16:41:30.430Z,
    "type": 5,
    "version": 1,
  },
}
`;

exports[`Reissue transaction service mget fetches real txs with nulls for unreal 1`] = `
Object {
  "__type": "list",
  "data": Array [
    Object {
      "__type": "transaction",
      "data": Object {
        "assetId": "F7CrEp9tihJSxTAAv2Kai1QQWBdF2fp3tyiDMRuvtcQm",
        "fee": "1",
        "height": 263697,
        "id": "2JksHUA6E4ZQd351i3oxJq2Ts7tvMN7NNVfoduR1fFsk",
        "quantity": "500",
        "reissuable": false,
        "sender": "3P7HKYe6HB8eK7uXxrztEYUSzZQyyYRnWHg",
        "senderPublicKey": "9ASL55YRs78rW5hjRAgjyWww7QqP99HwuB8dG6AqtnFm",
        "signature": "3FsqtYU1gt51F4YyEpG43e9XntY3qkxUdUZzxMzvb28MpaxytE7pw22gafwe11Za3uLvDvWUcxnoHj1dqCtRwfhE",
        "timestamp": 2016-12-06T16:41:30.430Z,
        "type": 5,
        "version": 1,
      },
    },
    null,
    Object {
      "__type": "transaction",
      "data": Object {
        "assetId": "FrEWpLGMKuB2wuVHPm4g1GEaPhuTaM3WBpiBr3N5ns44",
        "fee": "1",
        "height": 264684,
        "id": "2TyaLLCzjvC3tVTWebq5hAvEakXeACRrf3Bgu4e8fSos",
        "quantity": "1",
        "reissuable": false,
        "sender": "3PCgo6hYPPs9eWY3MBABEaR74KDHk7GfLxo",
        "senderPublicKey": "8kZrVHQBWmBqKKsu1jfgXoVhyL5b8REFTpMsmwnDLM1X",
        "signature": "2NY9uQ9sCJ3fLcsB26hB85KW9swXytnvejx1gfXRfCyXCurJ6mtzMVx63RGZ5GTwK6jUHZLxXeXFioVfQKuDEGAN",
        "timestamp": 2016-12-07T08:21:00.941Z,
        "type": 5,
        "version": 1,
      },
    },
  ],
}
`;
